### PR TITLE
Add sample

### PR DESCRIPTION
**Pull request by aschenkel**
@jpgarcia do not review this PR yet! since I look forward to adding some automatic testing using Detox, following the pattern [RNN playground](https://github.com/wix/react-native-navigation/blob/master/e2e/ApplicationLifecycleTest.test.js) is tested like. 
(except @ayelenmarie wants to make a collaboration since she's familiar with it 👀 😄 )

The example is working locally, so you can test a new change before updating the package on npm.

Went for a single example and not for several versioned samples because it seems cleaner & can focus on adding E2E tests to it.

---- 

It took me a while to send this PR since I had to solve some issues with my xcode and macOS system versions first

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -82,7 +82,7 @@
   "husky": {
     "hooks": {
       "prepare-commit-msg": "exec < /dev/tty && git cz --hook || true",
-      "pre-push": "yarn test"
+      "pre-push": "yarn lint && yarn test && cd example && yarn lint"
```
Comment (jpgarcia) [inline package.json:85]: You're running `yarn lint` twice

File: example/package.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  "private": true,
+  "scripts": {
+    "android": "react-native run-android",
+    "ios": "react-native run-ios",
+    "start": "react-native start",
+    "test": "jest",
+    "lint": "eslint ."
+  },
+  "dependencies": {
+    "prop-types": "^15.7.2",
```
Comment (jpgarcia) [inline example/package.json:13]: I'd set this as "15.7.2" also we could set `save-exact=true` in the `.npmrc` file

File: package.json (hunk 1 above)
Comment (aschenkel) [reply package.json:85]: the first one is for the lib itself & the second one is for the `example`.

`yarn lint` without `cd` into the example was not running it.

Comment (aschenkel) [reply package.json:85]: I have refactor the `lint` script to run `eslint` on both the lib & the project, and kept the pre-push neater.

**Comment by aschenkel** [comment]
ezgif-3-a3e493661c9b

Most events are tested automatically now!

As detailed on the [example's README](https://github.com/underscopeio/react-native-navigation-hooks/pull/28/files#diff-20bfc3ff6a860483887b93bf9cf0d135) there a couple of missing iOS +11 events tests

We should probably see to use something specific to run the E2E tests or configure `circleci` if possible (since that is the reason it is crashing)

There is also a minor detail that is when a detox test fail you have to scroll up through the console to see it because there is other long jest issue appearing. As I said its really minor, but worth mentioning.

File: package.json [hunk 3]
```diff
@@ -25,7 +25,7 @@
   "scripts": {
     "lint": "eslint *.js  && cd example && yarn lint",
     "build": "rm -rf ./dist && tsc",
-    "test": "jest",
+    "test": "jest src/",
```
Comment (aschenkel) [inline package.json:28]: I did this because otherwise example tests would run and you would need `detox` setted up for that

**Comment by aschenkel** [comment]
@jpgarcia I'm having some issues running the E2E detox tests on CircleCI and this PR is already a big one. Maybe its best if u can review it as it is now and I'll continue what I mentioned on another PR.

File: example/src/App.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+//           {
+//             component: {
+//               name: 'com.RNNHooksSample.MainScreen',
+//             },
+//           },
+//         ],
+//       },
+//     },
+//   })
+// })
```
Comment (ignacioola) [inline example/src/App.js:106]: Should you remove these comments?

**Pull request merged**